tools/test-npm.bat

```diff
@@ -0,0 +1,47 @@
+setlocal
+@rem Port of tools/test-npm.sh to windows
+
+@rem always change the working directory to the project's root directory
+cd %~dp0\..
+
+@rem pass a %NODE_EXE% variable from something like vcbuild.bat
+@rem it should point to Release\node.exe
+if not defined NODE_EXE (
+  echo No node executable provided. Bailing 1>&2
+  exit 0
+)
+
+rd /s /q test-npm
+del test-npm.tap
+
+@rem make a copy of deps/npm to run the tests on
+robocopy deps\npm test-npm /e >robocopy.log
+cd test-npm
+
+@rem do a rm first just in case deps/npm contained these
+rd /s /q npm-cache npm-tmp npm-prefix
+mkdir npm-cache npm-tmp npm-prefix
+
+@rem set some npm env variables to point to our new temporary folders
+set npm_config_cache=%cd%\npm-cache
+set npm_config_prefix=%cd%\npm-prefix
+set npm_config_tmp=%cd%\npm-tmp
+
+@rem ensure npm always uses the local node
+for /f "usebackq tokens=*" %%a in (`..\%NODE_EXE% -p "require('path').resolve('../Release')"`) do set PATH=%%a;%PATH%
+
+@rem make sure the binaries from the non-dev-deps are available
+node cli.js rebuild
+@rem install npm devDependencies and run npm's tests
+node cli.js install --ignore-scripts
+
+@rem run the tests
+if defined debug (
+  node_modules\.bin\tap.cmd --timeout 240 --reporter=tap test\tap
```

Comment:
Aaand, yeah, now that I paste that in it obviously _wouldn't_ work on Windows. Buuut, we're way better about ensuring that the version of node you ran npm with is the one used for lifecycle scripts, so we can actually rewrite that as:

```
    "test-node": "tap --timeout 240 \"test/tap/*.js\" \"test/network/*.js\""
```

which will work on windows.